### PR TITLE
Added module to manage InfraRsDomP class

### PR DESCRIPTION
**Pull request by sajagana**
Note:

Added module to manage Attachable Access Entity Profile - domain interfaces

### CODE REVIEW

File: aci/data_source_aci_infrarsdomp.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+				Optional: true,
+				Computed: true,
+			},
+		}),
+	}
+}
+
+func dataSourceAciInfraRsDomPRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	aciClient := m.(*client.Client)
+	tDn := d.Get("tDn").(string)
```
Comment (shrsr) [inline aci/data_source_aci_infrarsdomp.go:37]: Isn't this supposed to be t_dn?

File: website/docs/r/aci_aaep_to_domain.html.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Importing ##
+
+An existing Domain can be [imported][docs-import] into this resource via its Dn, via the following command:
+[docs-import]: https://www.terraform.io/docs/import/index.html
+
+
+```
+terraform import aci_domain.example <Dn>
```
Comment (shrsr) [inline website/docs/r/aci_aaep_to_domain.html.markdown:48]: terraform import aci_aaep_to_domain.example <Dn>

**Review by shrsr — COMMENTED** [review]
Please make changes.

**Review by anvitha-jain — APPROVED** [review]
LGTM

File: aci/resource_aci_infrarsdomp.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"annotation": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"t_dn": {
```
Comment (lhercot) [inline aci/resource_aci_infrarsdomp.go:36]: domain_dn

File: aci/resource_aci_infrarsdomp.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(map[string]*schema.Schema{
+			"attachable_access_entity_profile_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"annotation": {
```
Comment (lhercot) [inline aci/resource_aci_infrarsdomp.go:31]: Due to annotation being a optional / modifiable attribute, you will need to add an update function.

File: examples/aci_aaep_to_domain/main.tf [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+resource "aci_attachable_access_entity_profile" "fooattachable_access_entity_profile" {
+  name        = "aaep_test"
+  description = "attachable_access_entity_profile created while acceptance testing"
+
+}
+
+output "test_aaep_name" {
+  value = resource.aci_attachable_access_entity_profile.fooattachable_access_entity_profile.id
+}
```
Comment (lhercot) [inline examples/aci_aaep_to_domain/main.tf:24]: Those outputs are not needed in the example

File: examples/aci_aaep_to_domain/main.tf [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+resource "aci_l3_domain_profile" "fool3_domain_profile" {
+  name       = "l3out_domain_profile_test"
+  annotation = "example"
+  name_alias = "example"
+}
+
+output "test_l3out_domain_profile" {
+  value = resource.aci_l3_domain_profile.fool3_domain_profile.id
+}
```
Comment (lhercot) [inline examples/aci_aaep_to_domain/main.tf:34]: This one also not needed.

File: examples/aci_aaep_to_domain/main.tf [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+resource "aci_aaep_to_domain" "foo_aaep_to_domain" {
+  attachable_access_entity_profile_dn = resource.aci_attachable_access_entity_profile.fooattachable_access_entity_profile.id
+  t_dn                                = resource.aci_l3_domain_profile.fool3_domain_profile.id
+}
+
+output "test_aaep_to_domain" {
+  value = resource.aci_aaep_to_domain.foo_aaep_to_domain.id
+}
```
Comment (lhercot) [inline examples/aci_aaep_to_domain/main.tf:43]: This one also not needed.

File: examples/aci_aaep_to_domain/main.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+output "test_aaep_name" {
+  value = resource.aci_attachable_access_entity_profile.fooattachable_access_entity_profile.id
+}
+
+resource "aci_l3_domain_profile" "fool3_domain_profile" {
+  name       = "l3out_domain_profile_test"
+  annotation = "example"
+  name_alias = "example"
```
Comment (lhercot) [inline examples/aci_aaep_to_domain/main.tf:29]: remove those as they are not needed for this example.

File: website/docs/d/aaep_to_domain.html.markdown [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+
+## API Information ##
+
+* `Class` - infraRsDomP
+* `Distinguished Name` - uni/infra/attentp-{name}/rsdomP-[{tDn}]
+
+## GUI Information ##
+
+* `Location` - Attachable Access Entity Profile -> Domains Associated to Interfaces
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:22]: AAEPs are under Fabric -> Access Policies -> Policies -> Global -> AAEP

File: website/docs/r/aaep_to_domain.html.markdown [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+Manages ACI Attachable Access Entity Profile  - VMM, Physical or External domain interfaces.
+
+## API Information ##
+
+* `Class` - infraRsDomP
+* `Distinguished Name` - uni/infra/attentp-{name}/rsdomP-[{tDn}]
+
+## GUI Information ##
+
+* `Location` - Attachable Access Entity Profile -> Domains Associated to Interfaces
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:21]: AAEPs are under Fabric -> Access Policies -> Policies -> Global -> AAEP

File: website/docs/r/aaep_to_domain.html.markdown [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  t_dn                                = aci_l3_domain_profile.fool3_domain_profile.id
+}
+```
+
+## Argument Reference ##
+
+* `attachable_access_entity_profile_dn` - (Required) Distinguished name of the parent AttachableAccessEntityProfile object.
+* `annotation` - (Optional) Annotation of the object Domain.
+
+* `t_dn` - (Optional) Target-dn.The virtual machine manager domain profile.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:38]: domain_dn. The Distringuished Name of a domain.

**Review by lhercot — CHANGES_REQUESTED** [review]

File: website/docs/d/aaep_to_domain.html.markdown [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+layout: "aci"
+page_title: "ACI: aci_aaep_to_domain"
+sidebar_current: "docs-aci-data-aaep-to-domain"
+description: |-
+  Data source for ACI Attachable Access Entity Profile - VMM, Physical or External domain interfaces.
+---
+
+# aci_aaep_to_domain #
+
+Data source for ACI Attachable Access Entity Profile - VMM, Physical or External domain interfaces.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:12]: Data source for ACI Attachable Access Entity Profile (AAEP) to domain (VMM, Physical or External domain) relationships.

File: website/docs/d/aaep_to_domain.html.markdown [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## API Information ##
+
+* `Class` - infraRsDomP
+* `Distinguished Name` - uni/infra/attentp-{name}/rsdomP-[{tDn}]
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:18]: name -> aaep_name
tDn -> domain_dn

File: website/docs/d/aaep_to_domain.html.markdown [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+```hcl
+data "aci_aaep_to_domain" "foo_aaep_to_domain" {
+  attachable_access_entity_profile_dn = aci_attachable_access_entity_profile.fooattachable_access_entity_profile.id
+  domain_dn                           = aci_l3_domain_profile.fool3_domain_profile.id
+}
+```
+
+## Argument Reference ##
+
+* `attachable_access_entity_profile_dn` - (Required) Distinguished name of parent AttachableAccessEntityProfile object.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:37]: of the parent Attachable Access Entity Profile object.

File: website/docs/d/aaep_to_domain.html.markdown [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+* `annotation` - (Optional) Annotation of the object Domain.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:38]: not a necessary argument so can be removed from here and is documented in the attribute section.

File: website/docs/d/aaep_to_domain.html.markdown [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+* `domain_dn` - (Required) TDn of object Domain.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:39]: The Distinguished name of the domain object.

File: website/docs/d/aaep_to_domain.html.markdown [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+
+## Attribute Reference ##
+* `id` - Attribute id set to the Dn of the Domain.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:42]: of the Attachable AccessEntity Profile to Domain Relationship object

File: website/docs/d/aaep_to_domain.html.markdown [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+* `annotation` - (Optional) Annotation of object Domain.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:43]: of the Attachable AccessEntity Profile to Domain Relationship object

File: website/docs/d/aaep_to_domain.html.markdown [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+* `domain_dn` - (Optional) Target-dn. The virtual machine manager domain profile.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:44]: remove as already provided as argument

File: website/docs/d/aaep_to_domain.html.markdown [hunk 20]
```diff
@@ -0,0 +1,44 @@
+---
+subcategory: "Access Policies"
+layout: "aci"
+page_title: "ACI: aci_aaep_to_domain"
+sidebar_current: "docs-aci-data-aaep-to-domain"
+description: |-
+  Data source for ACI Attachable Access Entity Profile - VMM, Physical or External domain interfaces.
```
Comment (lhercot) [inline website/docs/d/aaep_to_domain.html.markdown:7]: Data source for ACI Attachable Access Entity Profile (AAEP) to domain (VMM, Physical or External domain) relationships.

File: website/docs/r/aaep_to_domain.html.markdown [hunk 21]
```diff
@@ -0,0 +1,48 @@
+---
+subcategory: "Access Policies"
+layout: "aci"
+page_title: "ACI: aci_aaep_to_domain"
+sidebar_current: "docs-aci-resource-aaep-to-domain"
+description: |-
+  Manages ACI Attachable Access Entity Profile - VMM, Physical or External domain interfaces.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:7]: Manages the ACI Attachable Access Entity Profile (AAEP) to domain (VMM, Physical or External domain) relationship

File: website/docs/r/aaep_to_domain.html.markdown [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+# aci_aaep_to_domain #
+
+Manages ACI Attachable Access Entity Profile  - VMM, Physical or External domain interfaces.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:12]: Manages the ACI Attachable Access Entity Profile (AAEP) to domain (VMM, Physical or External domain) relationship

File: website/docs/r/aaep_to_domain.html.markdown [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+
+## API Information ##
+
+* `Class` - infraRsDomP
+* `Distinguished Name` - uni/infra/attentp-{name}/rsdomP-[{tDn}]
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:17]: name -> aaep_name
tDn -> domain_dn

File: website/docs/r/aaep_to_domain.html.markdown [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+```hcl
+resource "aci_aaep_to_domain" "foo_aaep_to_domain" {
+  attachable_access_entity_profile_dn = aci_attachable_access_entity_profile.fooattachable_access_entity_profile.id
+  domain_dn                           = aci_l3_domain_profile.fool3_domain_profile.id
+}
+```
+
+## Argument Reference ##
+
+* `attachable_access_entity_profile_dn` - (Required) Distinguished name of the parent AttachableAccessEntityProfile object.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:35]: of the parent Attachable Access Entity Profile object.

File: website/docs/r/aaep_to_domain.html.markdown [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+* `annotation` - (Optional) Annotation of the object Domain.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:36]: of the Attachable AccessEntity Profile to Domain Relationship object

File: website/docs/r/aaep_to_domain.html.markdown [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+* `domain_dn` - (Optional) Target-dn.The virtual machine manager domain profile.
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:37]: (Required) The Distinguished name of the domain object.

File: website/docs/r/aaep_to_domain.html.markdown [hunk 27]
```diff
@@ -0,0 +1,48 @@
...
+
+
+## Importing ##
+
+An existing Domain can be [imported][docs-import] into this resource via its Dn, via the following command:
```
Comment (lhercot) [inline website/docs/r/aaep_to_domain.html.markdown:42]: An existing Attachable Access Entity Profile to Domain Relationship object can ...

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — CHANGES_REQUESTED** [review]
Also don't forget to get the latest go client and commit the files to this PR.

```
go get github.com/ciscoecosystem/aci-go-client@v1.xx.y
go mod vendor
```

File: examples/aci_aaep_to_domain/main.tf [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+provider "aci" {
+  username = ""
+  password = ""
+  url      = ""
+  insecure = true
+}
+
+resource "aci_attachable_access_entity_profile" "fooattachable_access_entity_profile" {
```
Comment (anvitha-jain) [inline examples/aci_aaep_to_domain/main.tf:16]: change fooattachable_access_entity_profile -> foo_attachable_access_entity_profile

File: examples/aci_aaep_to_domain/main.tf [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+  name        = "aaep_test"
+  description = "attachable_access_entity_profile created while acceptance testing"
+}
+
+resource "aci_l3_domain_profile" "fool3_domain_profile" {
```
Comment (anvitha-jain) [inline examples/aci_aaep_to_domain/main.tf:21]: change fool3_domain_profile -> foo_l3_domain_profile

File: go.mod [hunk 30]
```diff
@@ -3,7 +3,9 @@ module github.com/terraform-providers/terraform-provider-aci
 go 1.12
 
 require (
-	github.com/ciscoecosystem/aci-go-client v1.25.0
+	github.com/ciscoecosystem/aci-go-client v1.26.2
 	github.com/ghodss/yaml v1.0.0
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.4.3
 )
+
+replace github.com/ciscoecosystem/aci-go-client => /Users/sajagana/go/src/github.com/ciscoecosystem/aci-go-client
```
Comment (anvitha-jain) [inline go.mod:11]: This line can be removed, it is only for local use.

File: go.sum [hunk 31]
```diff
@@ -67,6 +67,8 @@ github.com/chzyer/readline v0.0.0-20180603132655-2972be24d48e/go.mod h1:nSuG5e5P
 github.com/chzyer/test v0.0.0-20180213035817-a1ea475d72b1/go.mod h1:Q3SI9o4m/ZMnBNeIyt5eFwwo7qiLfzFZmjNmxjkiQlU=
 github.com/ciscoecosystem/aci-go-client v1.25.0 h1:mw6Yk3DkYRLv6Te8/a9fVHJXA1tPI13hFg9rLGHkx6g=
 github.com/ciscoecosystem/aci-go-client v1.25.0/go.mod h1:pGK3tDhR/D4+P5f6utT1V7TQQxDw2fL7j79QXeVk6po=
```
Comment (anvitha-jain) [inline go.sum:69]: Remove line 68 and 69, older versions are not required.

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

**Comment by lhercot** [comment]
You probably also just want to rebase it on latest master

**Review by lhercot — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Pull request merged**